src/helpers.ts

```diff
@@ -1,13 +1,44 @@
 import type {
   ColumnarVectors,
-  MakeRequestTiming,
   RequestTiming,
   Vector,
 } from "./types";
 
-export const runtime = globalThis.navigator?.userAgent.split("/")[0];
+type Runtime =
+  | "bun"
+  | "deno"
+  | "cloudflare-workers"
+  | "browser"
+  | "node"
+  | undefined;
+
+function detectRuntime(): Runtime {
+  // @ts-expect-error can be ignored
+  if (typeof globalThis.Bun !== "undefined")
+    return "bun";
+
+  // @ts-expect-error can be ignored
+  if (typeof globalThis.Deno !== "undefined")
+    return "deno";
+
+  // Look for presence of non-standard globals specific to the cloudflare runtime:
+  // https://community.cloudflare.com/t/how-to-detect-the-cloudflare-worker-runtime/293715/2.
+  // At some point in the future, we can move to using navigator.userAgent:
```

Comment:
if cloudflare or any other runtime reliably always has userAgent set we can use it

if navigator?.userAgent is falsy fall back to other detection

Reply:
using `globalThis` for Bun and Deno is reliable. But I've added the `navigator?.userAgent` checks for cloudflare and node